linking error on gcc

Hi just pulled newest ifcos. I used commit 625 https://github.com/IfcOpenShell/IfcOpenShell/commit/fd4086aa8d7b22eed9c9e6535007f8c66e9bf0e7

I get tons of linking errors on ifcconvert:

~~~
hugo@weide:~/Documents/dev/ifcopenshell/ifcopenshell-official/build$ cmake   \
> -DOCC_INCLUDE_DIR=/usr/include/oce  \
> -DPYTHON_INCLUDE_DIR=/usr/include/python2.7  \
> -DPYTHON_LIBRARY=/usr/lib/python2.7/config-x86_64-linux-gnu/libpython2.7.so  \
> -DUNICODE_SUPPORT=True  \
> -DCOLLADA_SUPPORT=False \
> -DUSE_IFC4=False  \
> -DBUILD_IFCPYTHON=True  \
> -DBUILD_EXAMPLES=False  \
> ../ifcos/cmake/
-- The C compiler identification is GNU 4.9.2
-- The CXX compiler identification is GNU 4.9.2
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- BINDIR: /usr/local/bin
-- INCLUDEDIR: /usr/local/include
-- LIBDIR: /usr/local/lib
-- Boost version: 1.55.0
-- Found the following Boost libraries:
--   system
--   program_options
--   regex
--   thread
--   date_time
-- Boost include files found in /usr/include
-- Boost libraries found in /usr/lib/x86_64-linux-gnu
-- Looking for opencascade include files in: /usr/include/oce
-- Header files found
-- Looking for opencascade library files in: /usr/lib
-- Use OCC_LIBRARY_DIR to specify another directory
-- Library files found
-- No ICU include directory specified
-- No ICU library directory specified
-- ICU libraries found
-- Looking for C++ include limits                                                                                                                                                  
-- Looking for C++ include limits - found                                                                                                                                          
-- Looking for C++ include climits                                                                                                                                                 
-- Looking for C++ include climits - found                                                                                                                                         
-- Looking for C++ include limits.h                                                                                                                                                
-- Looking for C++ include limits.h - found                                                                                                                                        
-- Looking for C++ include fstream                                                                                                                                                 
-- Looking for C++ include fstream - found                                                                                                                                         
-- Looking for C++ include fstream.h                                                                                                                                               
-- Looking for C++ include fstream.h - not found                                                                                                                                   
-- Looking for C++ include iomanip                                                                                                                                                 
-- Looking for C++ include iomanip - found                                                                                                                                         
-- Looking for C++ include iomanip.h                                                                                                                                               
-- Looking for C++ include iomanip.h - not found                                                                                                                                   
-- Looking for C++ include iostream                                                                                                                                                
-- Looking for C++ include iostream - found                                                                                                                                        
-- Looking for C++ include iostream.h                                                                                                                                              
-- Looking for C++ include iostream.h - not found                                                                                                                                  
-- Set INSTALL_RPATH for IfcConvert: /usr/lib/x86_64-linux-gnu                                                                                                                     
-- Set INSTALL_RPATH for IfcGeomServer: /usr/lib/x86_64-linux-gnu                                                                                                                  
-- Found SWIG: /usr/bin/swig2.0 (found version "2.0.12")                                                                                                                           
-- Found PythonLibs: /usr/lib/python2.7/config-x86_64-linux-gnu/libpython2.7.so (found version "2.7.9")                                                                            
-- Set INSTALL_RPATH for _ifcopenshell_wrapper:                                                                                                                                    
-- Found PythonInterp: /usr/bin/python (found version "2.7.9")                                                                                                                     
-- Configuring done                                                                                                                                                                
-- Generating done                                                                                                                                                                 
-- Build files have been written to: /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/build                                                                             
hugo@weide:~/Documents/dev/ifcopenshell/ifcopenshell-official/build$ make                                                                                                          
Scanning dependencies of target IfcParse                                                                                                                                           
[  3%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcparse/IfcUtil.cpp.o                                     
[  6%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcparse/IfcHierarchyHelper.cpp.o                          
[  9%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcparse/IfcSpfHeader.cpp.o                                
[ 12%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcparse/IfcLogger.cpp.o                                   
[ 15%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcparse/IfcWrite.cpp.o                                    
[ 18%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcparse/Ifc2x3-latebound.cpp.o                            
[ 21%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcparse/IfcSIPrefix.cpp.o                                 
[ 24%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcparse/IfcCharacterDecoder.cpp.o                         
[ 27%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcparse/Ifc2x3.cpp.o                                      
[ 30%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcparse/IfcLateBoundEntity.cpp.o                          
[ 33%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcparse/IfcGlobalId.cpp.o                                 
[ 36%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcparse/IfcParse.cpp.o                                    
Linking CXX static library libIfcParse.a                                                                                                                                           
[ 36%] Built target IfcParse                                                                                                                                                       
Scanning dependencies of target IfcGeom                                                                                                                                            
[ 39%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcgeom/IfcGeomSerialisation.cpp.o                          
[ 42%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcgeom/IfcGeomCurves.cpp.o                                 
[ 45%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcgeom/IfcGeomShapes.cpp.o                                 
[ 48%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcgeom/IfcGeomHelpers.cpp.o                                
[ 51%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcgeom/IfcGeomRenderStyles.cpp.o                           
[ 54%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcgeom/IfcGeomWires.cpp.o                                  
[ 57%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcgeom/IfcGeomFunctions.cpp.o                              
/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcgeom/IfcGeomFunctions.cpp:139:2: warning: #warning "You are linking against linking against an older version of Open CASCADE. Version 6.9.0 introduces various improvements with relation to boolean operations. You are advised to upgrade." [-Wcpp]                                        
 #warning "You are linking against linking against an older version of Open CASCADE. Version 6.9.0 introduces various improvements with relation to boolean operations. You are advised to upgrade."                                                                                                                                                                  
  ^                                                                                                                                                                                
[ 60%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcgeom/IfcGeomRepresentation.cpp.o                         
[ 63%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcgeom/IfcGeomFaces.cpp.o                                  
[ 66%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcgeom/IfcRegister.cpp.o                                   
[ 69%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcgeom/IfcGeomMaterial.cpp.o                               
Linking CXX static library libIfcGeom.a                                                                                                                                            
[ 69%] Built target IfcGeom                                                                                                                                                        
Scanning dependencies of target IfcConvert                                                                                                                                         
[ 72%] Building CXX object CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcconvert/ColladaSerializer.cpp.o                       
[ 75%] Building CXX object CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcconvert/XmlSerializer.cpp.o                           
[ 78%] Building CXX object CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcconvert/util.cpp.o                                    
[ 81%] Building CXX object CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcconvert/SvgSerializer.cpp.o                           
[ 84%] Building CXX object CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcconvert/IfcConvert.cpp.o
[ 87%] Building CXX object CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcconvert/OpenCascadeBasedSerializer.cpp.o
[ 90%] Building CXX object CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcconvert/WavefrontObjSerializer.cpp.o
Linking CXX executable IfcConvert
CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-official/ifcos/src/ifcconvert/SvgSerializer.cpp.o: In function `SvgSerializer::write(std::pair<std::string, std::vector<util::string_buffer, std::allocator<util::string_buffer> > >&, TopoDS_Wire const&)':
SvgSerializer.cpp:(.text+0x2349): undefined reference to `Geom_Ellipse_Type_()'
SvgSerializer.cpp:(.text+0x2552): undefined reference to `Handle_Standard_Transient::EndScope()'
SvgSerializer.cpp:(.text+0x255c): undefined reference to `Handle_Standard_Transient::EndScope()'
SvgSerializer.cpp:(.text+0x2761): undefined reference to `Handle_Standard_Transient::BeginScope()'
SvgSerializer.cpp:(.text+0x2766): undefined reference to `Geom_Circle_Type_()'
SvgSerializer.cpp:(.text+0x2791): undefined reference to `Handle_Geom_Conic::DownCast(Handle_Standard_Transient const&)'
SvgSerializer.cpp:(.text+0x280f): undefined reference to `Geom_Circle_Type_()'
SvgSerializer.cpp:(.text+0x2835): undefined reference to `Handle_Geom_Ellipse::DownCast(Handle_Standard_Transient const&)'
SvgSerializer.cpp:(.text+0x2869): undefined reference to `Handle_Standard_Transient::EndScope()'
SvgSerializer.cpp:(.text+0x3541): undefined reference to `Handle_Standard_Transient::EndScope()'
SvgSerializer.cpp:(.text+0x3cb1): undefined reference to `Handle_Geom_Circle::DownCast(Handle_Standard_Transient const&)'
SvgSerializer.cpp:(.text+0x3cd8): undefined reference to `Handle_Standard_Transient::EndScope()'
SvgSerializer.cpp:(.text+0x42f2): undefined reference to `TopLoc_SListOfItemLocation::Clear()'
SvgSerializer.cpp:(.text+0x42fb): undefined reference to `Handle_Standard_Transient::EndScope()'
SvgSerializer.cpp:(.text+0x4304): undefined reference to `Handle_Standard_Transient::EndScope()'

IfcGeomFaces.cpp:(.text._ZN24BRepOffsetAPI_MakeOffsetD0Ev[_ZN24BRepOffsetAPI_MakeOffsetD5Ev]+0xa0): undefined reference to `BRepBuilderAPI_Command::Delete()'
IfcGeomFaces.cpp:(.text._ZN24BRepOffsetAPI_MakeOffsetD0Ev[_ZN24BRepOffsetAPI_MakeOffsetD5Ev]+0xaf): undefined reference to `Standard::Free(void*&)'
IfcGeomFaces.cpp:(.text._ZN24BRepOffsetAPI_MakeOffsetD0Ev[_ZN24BRepOffsetAPI_MakeOffsetD5Ev]+0xd0): undefined reference to `BRepFill_ListOfOffsetWire::Clear()'
IfcGeomFaces.cpp:(.text._ZN24BRepOffsetAPI_MakeOffsetD0Ev[_ZN24BRepOffsetAPI_MakeOffsetD5Ev]+0xd9): undefined reference to `TopTools_ListOfShape::Clear()'
IfcGeomFaces.cpp:(.text._ZN24BRepOffsetAPI_MakeOffsetD0Ev[_ZN24BRepOffsetAPI_MakeOffsetD5Ev]+0xfd): undefined reference to `Handle_Standard_Transient::EndScope()'
IfcGeomFaces.cpp:(.text._ZN24BRepOffsetAPI_MakeOffsetD0Ev[_ZN24BRepOffsetAPI_MakeOffsetD5Ev]+0x106): undefined reference to `Handle_Standard_Transient::EndScope()'
IfcGeomFaces.cpp:(.text._ZN24BRepOffsetAPI_MakeOffsetD0Ev[_ZN24BRepOffsetAPI_MakeOffsetD5Ev]+0x11c): undefined reference to `BRepBuilderAPI_Command::Delete()'
IfcGeomFaces.cpp:(.text._ZN24BRepOffsetAPI_MakeOffsetD0Ev[_ZN24BRepOffsetAPI_MakeOffsetD5Ev]+0x137): undefined reference to `TopTools_ListOfShape::Clear()'
IfcGeomFaces.cpp:(.text._ZN24BRepOffsetAPI_MakeOffsetD0Ev[_ZN24BRepOffsetAPI_MakeOffsetD5Ev]+0x15f): undefined reference to `Handle_Standard_Transient::EndScope()'
IfcGeomFaces.cpp:(.text._ZN24BRepOffsetAPI_MakeOffsetD0Ev[_ZN24BRepOffsetAPI_MakeOffsetD5Ev]+0x168): undefined reference to `Handle_Standard_Transient::EndScope()'
collect2: error: ld returned 1 exit status
CMakeFiles/IfcConvert.dir/build.make:250: recipe for target 'IfcConvert' failed
make[2]: *** [IfcConvert] Error 1
CMakeFiles/Makefile2:61: recipe for target 'CMakeFiles/IfcConvert.dir/all' failed
make[1]: *** [CMakeFiles/IfcConvert.dir/all] Error 2
Makefile:117: recipe for target 'all' failed
make: *** [all] Error 2
~~~

my cmake:
https://github.com/berndhahnebach/IfcOpenShell/commit/747bb9db57594b92415bb8328b43ed5f1e417035

~~~
 cmake                                                   \
> -DOCC_INCLUDE_DIR=/usr/include/oce                      \
> -DOCC_LIBRARY_DIR=/usr/lib/x86_64-linux-gnu/oce-0.15    \
> -DlibTKernel=/usr/lib/x86_64-linux-gnu/libTKernel.so    \
> -DPYTHON_INCLUDE_DIR=/usr/include/python2.7             \
> -DPYTHON_LIBRARY=/usr/lib/python2.7/config-x86_64-linux-gnu/libpython2.7.so  \
> -DUNICODE_SUPPORT=True  \
> -DCOLLADA_SUPPORT=False \
> -DUSE_IFC4=False        \
> -DBUILD_IFCPYTHON=True  \
> -DBUILD_EXAMPLES=False  \
> ../ifcos/cmake/
-- The C compiler identification is GNU 4.9.2
-- The CXX compiler identification is GNU 4.9.2
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- BINDIR: /usr/local/bin
-- INCLUDEDIR: /usr/local/include
-- LIBDIR: /usr/local/lib
-- Boost version: 1.55.0
-- Found the following Boost libraries:
--   system
--   program_options
--   regex
--   thread
--   date_time
-- Boost include files found in /usr/include
-- Boost libraries found in /usr/lib/x86_64-linux-gnu                                                                                                                                                                  
-- Looking for opencascade include files in: /usr/include/oce
-- Header files found
-- OCC_INCLUDE_DIR /usr/include/oce
-- Looking for opencascade library files in: /usr/lib/x86_64-linux-gnu/oce-0.15
-- Library files found
-- OCC_LIBRARY_DIR /usr/lib/x86_64-linux-gnu/oce-0.15
-- libTKernel /usr/lib/x86_64-linux-gnu/libTKernel.so
-- No ICU include directory specified
-- No ICU library directory specified
-- ICU libraries found
-- Looking for C++ include limits
-- Looking for C++ include limits - found
-- Looking for C++ include climits
-- Looking for C++ include climits - found
-- Looking for C++ include limits.h
-- Looking for C++ include limits.h - found
-- Looking for C++ include fstream
-- Looking for C++ include fstream - found
-- Looking for C++ include fstream.h
-- Looking for C++ include fstream.h - not found
-- Looking for C++ include iomanip
-- Looking for C++ include iomanip - found
-- Looking for C++ include iomanip.h
-- Looking for C++ include iomanip.h - not found
-- Looking for C++ include iostream
-- Looking for C++ include iostream - found
-- Looking for C++ include iostream.h
-- Looking for C++ include iostream.h - not found
-- LINK_DIRECTORIES 
-- Set INSTALL_RPATH for IfcConvert: /usr/lib/x86_64-linux-gnu/oce-0.15;/usr/lib/x86_64-linux-gnu
-- Set INSTALL_RPATH for IfcGeomServer: /usr/lib/x86_64-linux-gnu/oce-0.15;/usr/lib/x86_64-linux-gnu
-- Found SWIG: /usr/bin/swig2.0 (found version "2.0.12") 
-- Found PythonLibs: /usr/lib/python2.7/config-x86_64-linux-gnu/libpython2.7.so (found version "2.7.9") 
-- Set INSTALL_RPATH for _ifcopenshell_wrapper: /usr/lib/x86_64-linux-gnu/oce-0.15
-- Found PythonInterp: /usr/bin/python (found version "2.7.9") 
-- Configuring done
-- Generating done
-- Build files have been written to: /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build
hugo@weide:~/Documents/dev/ifcopenshell/ifcopenshell-bhb/build$ make
Scanning dependencies of target IfcParse
[  3%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcparse/IfcUtil.cpp.o
[  6%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcparse/IfcHierarchyHelper.cpp.o
[  9%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcparse/IfcSpfHeader.cpp.o
[ 12%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcparse/IfcLogger.cpp.o
[ 15%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcparse/IfcWrite.cpp.o
[ 18%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcparse/Ifc2x3-latebound.cpp.o
[ 21%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcparse/IfcSIPrefix.cpp.o
[ 24%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcparse/IfcCharacterDecoder.cpp.o
[ 27%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcparse/Ifc2x3.cpp.o
[ 30%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcparse/IfcLateBoundEntity.cpp.o
[ 33%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcparse/IfcGlobalId.cpp.o
[ 36%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcparse/IfcParse.cpp.o
Linking CXX static library libIfcParse.a
[ 36%] Built target IfcParse
Scanning dependencies of target IfcGeom
[ 39%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcgeom/IfcGeomSerialisation.cpp.o
[ 42%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcgeom/IfcGeomCurves.cpp.o
[ 45%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcgeom/IfcGeomShapes.cpp.o
[ 48%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcgeom/IfcGeomHelpers.cpp.o
[ 51%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcgeom/IfcGeomRenderStyles.cpp.o
[ 54%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcgeom/IfcGeomWires.cpp.o
[ 57%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcgeom/IfcGeomFunctions.cpp.o
/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcgeom/IfcGeomFunctions.cpp:139:2: warning: #warning "You are linking against linking against an older version of Open CASCADE. Version 6.9.0 introduces various improvements with relation to boolean operations. You are advised to upgrade." [-Wcpp]
 #warning "You are linking against linking against an older version of Open CASCADE. Version 6.9.0 introduces various improvements with relation to boolean operations. You are advised to upgrade."
  ^
[ 60%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcgeom/IfcGeomRepresentation.cpp.o
[ 63%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcgeom/IfcGeomFaces.cpp.o
[ 66%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcgeom/IfcRegister.cpp.o
[ 69%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcgeom/IfcGeomMaterial.cpp.o
Linking CXX static library libIfcGeom.a
[ 69%] Built target IfcGeom
Scanning dependencies of target IfcConvert
[ 72%] Building CXX object CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/ColladaSerializer.cpp.o
[ 75%] Building CXX object CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/XmlSerializer.cpp.o
[ 78%] Building CXX object CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/util.cpp.o
[ 81%] Building CXX object CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/SvgSerializer.cpp.o
[ 84%] Building CXX object CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/IfcConvert.cpp.o
[ 87%] Building CXX object CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/OpenCascadeBasedSerializer.cpp.o
[ 90%] Building CXX object CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/WavefrontObjSerializer.cpp.o
Linking CXX executable IfcConvert
CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/SvgSerializer.cpp.o: In function `SvgSerializer::write(std::pair<std::string, std::vector<util::string_buffer, std::allocator<util::string_buffer> > >&, TopoDS_Wire const&)':
SvgSerializer.cpp:(.text+0x2349): undefined reference to `Geom_Ellipse_Type_()'
SvgSerializer.cpp:(.text+0x2552): undefined reference to `Handle_Standard_Transient::EndScope()'
SvgSerializer.cpp:(.text+0x255c): undefined reference to `Handle_Standard_Transient::EndScope()'
...
IfcGeomFaces.cpp:(.text._ZN24BRepOffsetAPI_MakeOffsetD0Ev[_ZN24BRepOffsetAPI_MakeOffsetD5Ev]+0x106): undefined reference to `Handle_Standard_Transient::EndScope()'
IfcGeomFaces.cpp:(.text._ZN24BRepOffsetAPI_MakeOffsetD0Ev[_ZN24BRepOffsetAPI_MakeOffsetD5Ev]+0x11c): undefined reference to `BRepBuilderAPI_Command::Delete()'
IfcGeomFaces.cpp:(.text._ZN24BRepOffsetAPI_MakeOffsetD0Ev[_ZN24BRepOffsetAPI_MakeOffsetD5Ev]+0x137): undefined reference to `TopTools_ListOfShape::Clear()'
IfcGeomFaces.cpp:(.text._ZN24BRepOffsetAPI_MakeOffsetD0Ev[_ZN24BRepOffsetAPI_MakeOffsetD5Ev]+0x15f): undefined reference to `Handle_Standard_Transient::EndScope()'
IfcGeomFaces.cpp:(.text._ZN24BRepOffsetAPI_MakeOffsetD0Ev[_ZN24BRepOffsetAPI_MakeOffsetD5Ev]+0x168): undefined reference to `Handle_Standard_Transient::EndScope()'
collect2: error: ld returned 1 exit status
CMakeFiles/IfcConvert.dir/build.make:250: recipe for target 'IfcConvert' failed
make[2]: *** [IfcConvert] Error 1
CMakeFiles/Makefile2:61: recipe for target 'CMakeFiles/IfcConvert.dir/all' failed
make[1]: *** [CMakeFiles/IfcConvert.dir/all] Error 2
Makefile:117: recipe for target 'all' failed
make: *** [all] Error 2
~~~

~~~
$ make VERBOSE=1
/usr/bin/cmake -H/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/cmake -B/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build --check-build-system CMakeFiles/Makefile.cmake 0
/usr/bin/cmake -E cmake_progress_start /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/CMakeFiles /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/CMakeFiles/progress.marks
make -f CMakeFiles/Makefile2 all
make[1]: Entering directory '/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build'
make -f CMakeFiles/IfcParse.dir/build.make CMakeFiles/IfcParse.dir/depend
make[2]: Entering directory '/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build'
cd /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/cmake /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/cmake /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/CMakeFiles/IfcParse.dir/DependInfo.cmake --color=
make[2]: Leaving directory '/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build'
make -f CMakeFiles/IfcParse.dir/build.make CMakeFiles/IfcParse.dir/build
make[2]: Entering directory '/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build'
make[2]: Nothing to be done for 'CMakeFiles/IfcParse.dir/build'.
make[2]: Leaving directory '/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build'
/usr/bin/cmake -E cmake_progress_report /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/CMakeFiles  20 21 22 23 24 25 26 27 28 29 30 31
[ 36%] Built target IfcParse
make -f CMakeFiles/IfcGeom.dir/build.make CMakeFiles/IfcGeom.dir/depend
make[2]: Entering directory '/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build'
cd /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/cmake /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/cmake /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/CMakeFiles/IfcGeom.dir/DependInfo.cmake --color=
make[2]: Leaving directory '/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build'
make -f CMakeFiles/IfcGeom.dir/build.make CMakeFiles/IfcGeom.dir/build
make[2]: Entering directory '/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build'
make[2]: Nothing to be done for 'CMakeFiles/IfcGeom.dir/build'.
make[2]: Leaving directory '/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build'
/usr/bin/cmake -E cmake_progress_report /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/CMakeFiles  8 9 10 11 12 13 14 15 16 17 18
[ 69%] Built target IfcGeom
make -f CMakeFiles/IfcConvert.dir/build.make CMakeFiles/IfcConvert.dir/depend
make[2]: Entering directory '/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build'
cd /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/cmake /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/cmake /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/CMakeFiles/IfcConvert.dir/DependInfo.cmake --color=
make[2]: Leaving directory '/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build'
make -f CMakeFiles/IfcConvert.dir/build.make CMakeFiles/IfcConvert.dir/build
make[2]: Entering directory '/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build'
Linking CXX executable IfcConvert                                                                                                                                                                                      
/usr/bin/cmake -E cmake_link_script CMakeFiles/IfcConvert.dir/link.txt --verbose=1
/usr/bin/c++   -O3 -DNDEBUG    CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/ColladaSerializer.cpp.o CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/XmlSerializer.cpp.o CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/util.cpp.o CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/SvgSerializer.cpp.o CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/IfcConvert.cpp.o CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/OpenCascadeBasedSerializer.cpp.o CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/WavefrontObjSerializer.cpp.o  -o IfcConvert  -L/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build  -L/usr/lib/x86_64-linux-gnu/oce-0.15  -L/usr/local/lib  -L/usr/local/lib64 -rdynamic libIfcParse.a libIfcGeom.a -lTKernel -lTKMath -lTKBRep -lTKGeomBase -lTKGeomAlgo -lTKG3d -lTKG2d -lTKShHealing -lTKTopAlgo -lTKMesh -lTKPrim -lTKBool -lTKBO -lTKFillet -lTKSTEP -lTKSTEPBase -lTKSTEPAttr -lTKXSBase -lTKSTEP209 -lTKIGES -lTKOffset -lboost_system -lboost_program_options -lboost_regex -lboost_thread -lboost_date_time -lpthread -licuuc -licudata -ldl libIfcParse.a -lboost_system -lboost_program_options -lboost_regex -lboost_thread -lboost_date_time -lpthread -licuuc -licudata -ldl -Wl,-rpath,/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build:/usr/lib/x86_64-linux-gnu/oce-0.15:/usr/local/lib:/usr/local/lib64: 
CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/SvgSerializer.cpp.o: In function `SvgSerializer::write(std::pair<std::string, std::vector<util::string_buffer, std::allocator<util::string_buffer> > >&, TopoDS_Wire const&)':
SvgSerializer.cpp:(.text+0x2349): undefined reference to `Geom_Ellipse_Type_()'
SvgSerializer.cpp:(.text+0x2552): undefined reference to `Handle_Standard_Transient::EndScope()'
SvgSerializer.cpp:(.text+0x255c): undefined reference to `Handle_Standard_Transient::EndScope()'
SvgSerializer.cpp:(.text+0x2761): undefined reference to `Handle_Standard_Transient::BeginScope()'
SvgSerializer.cpp:(.text+0x2766): undefined reference to `Geom_Circle_Type_()'
SvgSerializer.cpp:(.text+0x2791): undefined reference to `Handle_Geom_Conic::DownCast(Handle_Standard_Transient const&)'
SvgSerializer.cpp:(.text+0x280f): undefined reference to `Geom_Circle_Type_()'
SvgSerializer.cpp:(.text+0x2835): undefined reference to `Handle_Geom_Ellipse::DownCast(Handle_Standard_Transient const&)'
SvgSerializer.cpp:(.text+0x2869): undefined reference to `Handle_Standard_Transient::EndScope()'
SvgSerializer.cpp:(.text+0x3541): undefined reference to `Handle_Standard_Transient::EndScope()'
SvgSerializer.cpp:(.text+0x3cb1): undefined reference to `Handle_Geom_Circle::DownCast(Handle_Standard_Transient const&)'
SvgSerializer.cpp:(.text+0x3cd8): undefined reference to `Handle_Standard_Transient::EndScope()'
SvgSerializer.cpp:(.text+0x42f2): undefined reference to `TopLoc_SListOfItemLocation::Clear()'
SvgSerializer.cpp:(.text+0x42fb): undefined reference to `Handle_Standard_Transient::EndScope()'
SvgSerializer.cpp:(.text+0x4304): undefined reference to `Handle_Standard_Transient::EndScope()'
...
fcGeomFaces.cpp:(.text._ZN24BRepOffsetAPI_MakeOffsetD0Ev[_ZN24BRepOffsetAPI_MakeOffsetD5Ev]+0x11c): undefined reference to `BRepBuilderAPI_Command::Delete()'
IfcGeomFaces.cpp:(.text._ZN24BRepOffsetAPI_MakeOffsetD0Ev[_ZN24BRepOffsetAPI_MakeOffsetD5Ev]+0x137): undefined reference to `TopTools_ListOfShape::Clear()'
IfcGeomFaces.cpp:(.text._ZN24BRepOffsetAPI_MakeOffsetD0Ev[_ZN24BRepOffsetAPI_MakeOffsetD5Ev]+0x15f): undefined reference to `Handle_Standard_Transient::EndScope()'
IfcGeomFaces.cpp:(.text._ZN24BRepOffsetAPI_MakeOffsetD0Ev[_ZN24BRepOffsetAPI_MakeOffsetD5Ev]+0x168): undefined reference to `Handle_Standard_Transient::EndScope()'
collect2: error: ld returned 1 exit status
CMakeFiles/IfcConvert.dir/build.make:250: recipe for target 'IfcConvert' failed
make[2]: *** [IfcConvert] Error 1
make[2]: Leaving directory '/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build'
CMakeFiles/Makefile2:61: recipe for target 'CMakeFiles/IfcConvert.dir/all' failed
make[1]: *** [CMakeFiles/IfcConvert.dir/all] Error 2
make[1]: Leaving directory '/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build'
Makefile:117: recipe for target 'all' failed
make: *** [all] Error 2
hugo@weide:~/Documents/dev/ifcopenshell/ifcopenshell-bhb/build$

~~~
hugo@weide:~/Documents/dev/ifcopenshell/ifcopenshell-bhb/build$ 
hugo@weide:~/Documents/dev/ifcopenshell/ifcopenshell-bhb/build$ rm -rf *
hugo@weide:~/Documents/dev/ifcopenshell/ifcopenshell-bhb/build$ 
hugo@weide:~/Documents/dev/ifcopenshell/ifcopenshell-bhb/build$ export CXXFLAGS="$CXXFLAGS -Wl,--verbose"
hugo@weide:~/Documents/dev/ifcopenshell/ifcopenshell-bhb/build$ export   CFLAGS="$CFLAGS   -Wl,--verbose"
hugo@weide:~/Documents/dev/ifcopenshell/ifcopenshell-bhb/build$ export  LDFLAGS="$LDFLAGS  -Wl,--verbose"
hugo@weide:~/Documents/dev/ifcopenshell/ifcopenshell-bhb/build$ 
hugo@weide:~/Documents/dev/ifcopenshell/ifcopenshell-bhb/build$ cmake                                                   \
> -DOCC_INCLUDE_DIR=/usr/include/oce                      \
> -DOCC_LIBRARY_DIR=/usr/lib/x86_64-linux-gnu             \
> -DlibTKernel=/usr/lib/x86_64-linux-gnu/libTKernel.so    \
> -DPYTHON_INCLUDE_DIR=/usr/include/python2.7             \
> -DPYTHON_LIBRARY=/usr/lib/python2.7/config-x86_64-linux-gnu/libpython2.7.so  \
> -DUNICODE_SUPPORT=True  \
> -DCOLLADA_SUPPORT=False \
> -DUSE_IFC4=False        \
> -DBUILD_IFCPYTHON=True  \
> -DBUILD_EXAMPLES=False  \
> ../ifcos/cmake/
-- The C compiler identification is GNU 4.9.2
-- The CXX compiler identification is GNU 4.9.2
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- BINDIR: /usr/local/bin
-- INCLUDEDIR: /usr/local/include
-- LIBDIR: /usr/local/lib
-- Boost version: 1.55.0
-- Found the following Boost libraries:
--   system
--   program_options
--   regex
--   thread
--   date_time
-- Boost include files found in /usr/include
-- Boost libraries found in /usr/lib/x86_64-linux-gnu                                                                                                                                                                  
-- Looking for opencascade include files in: /usr/include/oce
-- Header files found
-- Looking for opencascade library files in: /usr/lib/x86_64-linux-gnu
-- Library files found
-- No ICU include directory specified
-- No ICU library directory specified
-- ICU libraries found
-- Looking for C++ include limits
-- Looking for C++ include limits - found
-- Looking for C++ include climits
-- Looking for C++ include climits - found
-- Looking for C++ include limits.h
-- Looking for C++ include limits.h - found
-- Looking for C++ include fstream
-- Looking for C++ include fstream - found
-- Looking for C++ include fstream.h
-- Looking for C++ include fstream.h - not found
-- Looking for C++ include iomanip
-- Looking for C++ include iomanip - found
-- Looking for C++ include iomanip.h
-- Looking for C++ include iomanip.h - not found
-- Looking for C++ include iostream
-- Looking for C++ include iostream - found
-- Looking for C++ include iostream.h
-- Looking for C++ include iostream.h - not found
-- Set INSTALL_RPATH for IfcConvert: /usr/lib/x86_64-linux-gnu;/usr/lib/x86_64-linux-gnu
-- Set INSTALL_RPATH for IfcGeomServer: /usr/lib/x86_64-linux-gnu;/usr/lib/x86_64-linux-gnu
-- Found SWIG: /usr/bin/swig2.0 (found version "2.0.12") 
-- Found PythonLibs: /usr/lib/python2.7/config-x86_64-linux-gnu/libpython2.7.so (found version "2.7.9") 
-- Set INSTALL_RPATH for _ifcopenshell_wrapper: /usr/lib/x86_64-linux-gnu
-- Found PythonInterp: /usr/bin/python (found version "2.7.9") 
-- Configuring done
-- Generating done
-- Build files have been written to: /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build
hugo@weide:~/Documents/dev/ifcopenshell/ifcopenshell-bhb/build$ 
hugo@weide:~/Documents/dev/ifcopenshell/ifcopenshell-bhb/build$ 
hugo@weide:~/Documents/dev/ifcopenshell/ifcopenshell-bhb/build$ 
hugo@weide:~/Documents/dev/ifcopenshell/ifcopenshell-bhb/build$ make
Scanning dependencies of target IfcParse
[  3%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcparse/IfcUtil.cpp.o
[  6%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcparse/IfcHierarchyHelper.cpp.o
[  9%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcparse/IfcSpfHeader.cpp.o
[ 12%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcparse/IfcLogger.cpp.o
[ 15%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcparse/IfcWrite.cpp.o
[ 18%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcparse/Ifc2x3-latebound.cpp.o
[ 21%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcparse/IfcSIPrefix.cpp.o
[ 24%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcparse/IfcCharacterDecoder.cpp.o
[ 27%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcparse/Ifc2x3.cpp.o
[ 30%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcparse/IfcLateBoundEntity.cpp.o
[ 33%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcparse/IfcGlobalId.cpp.o
[ 36%] Building CXX object CMakeFiles/IfcParse.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcparse/IfcParse.cpp.o
Linking CXX static library libIfcParse.a
[ 36%] Built target IfcParse
Scanning dependencies of target IfcGeom
[ 39%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcgeom/IfcGeomSerialisation.cpp.o
[ 42%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcgeom/IfcGeomCurves.cpp.o
[ 45%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcgeom/IfcGeomShapes.cpp.o
[ 48%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcgeom/IfcGeomHelpers.cpp.o
[ 51%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcgeom/IfcGeomRenderStyles.cpp.o
[ 54%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcgeom/IfcGeomWires.cpp.o
[ 57%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcgeom/IfcGeomFunctions.cpp.o
/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcgeom/IfcGeomFunctions.cpp:139:2: warning: #warning "You are linking against linking against an older version of Open CASCADE. Version 6.9.0 introduces various improvements with relation to boolean operations. You are advised to upgrade." [-Wcpp]
 #warning "You are linking against linking against an older version of Open CASCADE. Version 6.9.0 introduces various improvements with relation to boolean operations. You are advised to upgrade."
  ^
[ 60%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcgeom/IfcGeomRepresentation.cpp.o
[ 63%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcgeom/IfcGeomFaces.cpp.o
[ 66%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcgeom/IfcRegister.cpp.o
[ 69%] Building CXX object CMakeFiles/IfcGeom.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcgeom/IfcGeomMaterial.cpp.o
Linking CXX static library libIfcGeom.a
[ 69%] Built target IfcGeom
Scanning dependencies of target IfcConvert
[ 72%] Building CXX object CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/ColladaSerializer.cpp.o
[ 75%] Building CXX object CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/XmlSerializer.cpp.o
[ 78%] Building CXX object CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/util.cpp.o
[ 81%] Building CXX object CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/SvgSerializer.cpp.o
[ 84%] Building CXX object CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/IfcConvert.cpp.o
[ 87%] Building CXX object CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/OpenCascadeBasedSerializer.cpp.o
[ 90%] Building CXX object CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/WavefrontObjSerializer.cpp.o
Linking CXX executable IfcConvert
GNU ld (GNU Binutils for Debian) 2.25
  Supported emulations:
   elf_x86_64
   elf32_x86_64
   elf_i386
   i386linux
   elf_l1om
   elf_k1om
   i386pep
   i386pe
GNU ld (GNU Binutils for Debian) 2.25
  Supported emulations:
   elf_x86_64
   elf32_x86_64
   elf_i386
   i386linux
   elf_l1om
   elf_k1om
   i386pep
   i386pe
using internal linker script:
==================================================
/* Script for -z combreloc: combine and sort reloc sections */
/* Copyright (C) 2014 Free Software Foundation, Inc.
   Copying and distribution of this script, with or without modification,
   are permitted in any medium without royalty provided the copyright
   notice and this notice are preserved.  */
OUTPUT_FORMAT("elf64-x86-64", "elf64-x86-64",
              "elf64-x86-64")
OUTPUT_ARCH(i386:x86-64)
ENTRY(_start)
SEARCH_DIR("=/usr/x86_64-linux-gnu/lib64"); SEARCH_DIR("=/usr/local/lib/x86_64-linux-gnu"); SEARCH_DIR("=/usr/local/lib64"); SEARCH_DIR("=/lib/x86_64-linux-gnu"); SEARCH_DIR("=/lib64"); SEARCH_DIR("=/usr/lib/x86_64-linux-gnu"); SEARCH_DIR("=/usr/lib64"); SEARCH_DIR("=/usr/x86_64-linux-gnu/lib"); SEARCH_DIR("=/usr/local/lib"); SEARCH_DIR("=/lib"); SEARCH_DIR("=/usr/lib");
SECTIONS
{
  /* Read-only sections, merged into text segment: */
  PROVIDE (__executable_start = SEGMENT_START("text-segment", 0x400000)); . = SEGMENT_START("text-segment", 0x400000) + SIZEOF_HEADERS;
  .interp         : { *(.interp) }
  .note.gnu.build-id : { *(.note.gnu.build-id) }
  .hash           : { *(.hash) }
  .gnu.hash       : { *(.gnu.hash) }
  .dynsym         : { *(.dynsym) }
  .dynstr         : { *(.dynstr) }
  .gnu.version    : { *(.gnu.version) }
  .gnu.version_d  : { *(.gnu.version_d) }
  .gnu.version_r  : { *(.gnu.version_r) }
  .rela.dyn       :
    {
      *(.rela.init)
      *(.rela.text .rela.text.* .rela.gnu.linkonce.t.*)
      *(.rela.fini)
      *(.rela.rodata .rela.rodata.* .rela.gnu.linkonce.r.*)
      *(.rela.data .rela.data.* .rela.gnu.linkonce.d.*)
      *(.rela.tdata .rela.tdata.* .rela.gnu.linkonce.td.*)
      *(.rela.tbss .rela.tbss.* .rela.gnu.linkonce.tb.*)
      *(.rela.ctors)
      *(.rela.dtors)
      *(.rela.got)
      *(.rela.bss .rela.bss.* .rela.gnu.linkonce.b.*)
      *(.rela.ldata .rela.ldata.* .rela.gnu.linkonce.l.*)
      *(.rela.lbss .rela.lbss.* .rela.gnu.linkonce.lb.*)
      *(.rela.lrodata .rela.lrodata.* .rela.gnu.linkonce.lr.*)
      *(.rela.ifunc)
    }
  .rela.plt       :
    {
      *(.rela.plt)
      PROVIDE_HIDDEN (__rela_iplt_start = .);
      *(.rela.iplt)
      PROVIDE_HIDDEN (__rela_iplt_end = .);
    }
  .init           :
  {
    KEEP (*(SORT_NONE(.init)))
  }
  .plt            : { *(.plt) *(.iplt) }
  .plt.bnd        : { *(.plt.bnd) }
  .text           :
  {
    *(.text.unlikely .text.*_unlikely .text.unlikely.*)
    *(.text.exit .text.exit.*)
    *(.text.startup .text.startup.*)
    *(.text.hot .text.hot.*)
    *(.text .stub .text.* .gnu.linkonce.t.*)
    /* .gnu.warning sections are handled specially by elf32.em.  */
    *(.gnu.warning)
  }
  .fini           :
  {
    KEEP (*(SORT_NONE(.fini)))
  }
  PROVIDE (__etext = .);
  PROVIDE (_etext = .);
  PROVIDE (etext = .);
  .rodata         : { *(.rodata .rodata.* .gnu.linkonce.r.*) }
  .rodata1        : { *(.rodata1) }
  .eh_frame_hdr : { *(.eh_frame_hdr) }
  .eh_frame       : ONLY_IF_RO { KEEP (*(.eh_frame)) }
  .gcc_except_table   : ONLY_IF_RO { *(.gcc_except_table
  .gcc_except_table.*) }
  /* These sections are generated by the Sun/Oracle C++ compiler.  */
  .exception_ranges   : ONLY_IF_RO { *(.exception_ranges
  .exception_ranges*) }
  /* Adjust the address for the data segment.  We want to adjust up to
     the same address within the page on the next page up.  */
  . = ALIGN (CONSTANT (MAXPAGESIZE)) - ((CONSTANT (MAXPAGESIZE) - .) & (CONSTANT (MAXPAGESIZE) - 1)); . = DATA_SEGMENT_ALIGN (CONSTANT (MAXPAGESIZE), CONSTANT (COMMONPAGESIZE));
  /* Exception handling  */
  .eh_frame       : ONLY_IF_RW { KEEP (*(.eh_frame)) }
  .gcc_except_table   : ONLY_IF_RW { *(.gcc_except_table .gcc_except_table.*) }
  .exception_ranges   : ONLY_IF_RW { *(.exception_ranges .exception_ranges*) }
  /* Thread Local Storage sections  */
  .tdata          : { *(.tdata .tdata.* .gnu.linkonce.td.*) }
  .tbss           : { *(.tbss .tbss.* .gnu.linkonce.tb.*) *(.tcommon) }
  .preinit_array     :
  {
    PROVIDE_HIDDEN (__preinit_array_start = .);
    KEEP (*(.preinit_array))
    PROVIDE_HIDDEN (__preinit_array_end = .);
  }
  .init_array     :
  {
    PROVIDE_HIDDEN (__init_array_start = .);
    KEEP (*(SORT_BY_INIT_PRIORITY(.init_array.*) SORT_BY_INIT_PRIORITY(.ctors.*)))
    KEEP (*(.init_array EXCLUDE_FILE (*crtbegin.o *crtbegin?.o *crtend.o *crtend?.o ) .ctors))
    PROVIDE_HIDDEN (__init_array_end = .);
  }
  .fini_array     :
  {
    PROVIDE_HIDDEN (__fini_array_start = .);
    KEEP (*(SORT_BY_INIT_PRIORITY(.fini_array.*) SORT_BY_INIT_PRIORITY(.dtors.*)))
    KEEP (*(.fini_array EXCLUDE_FILE (*crtbegin.o *crtbegin?.o *crtend.o *crtend?.o ) .dtors))
    PROVIDE_HIDDEN (__fini_array_end = .);
  }
  .ctors          :
  {
    /* gcc uses crtbegin.o to find the start of
       the constructors, so we make sure it is
       first.  Because this is a wildcard, it
       doesn't matter if the user does not
       actually link against crtbegin.o; the
       linker won't look for a file to match a
       wildcard.  The wildcard also means that it
       doesn't matter which directory crtbegin.o
       is in.  */
    KEEP (*crtbegin.o(.ctors))
    KEEP (*crtbegin?.o(.ctors))
    /* We don't want to include the .ctor section from
       the crtend.o file until after the sorted ctors.
       The .ctor section from the crtend file contains the
       end of ctors marker and it must be last */
    KEEP (*(EXCLUDE_FILE (*crtend.o *crtend?.o ) .ctors))
    KEEP (*(SORT(.ctors.*)))
    KEEP (*(.ctors))
  }
  .dtors          :
  {
    KEEP (*crtbegin.o(.dtors))
    KEEP (*crtbegin?.o(.dtors))
    KEEP (*(EXCLUDE_FILE (*crtend.o *crtend?.o ) .dtors))
    KEEP (*(SORT(.dtors.*)))
    KEEP (*(.dtors))
  }
  .jcr            : { KEEP (*(.jcr)) }
  .data.rel.ro : { *(.data.rel.ro.local* .gnu.linkonce.d.rel.ro.local.*) *(.data.rel.ro .data.rel.ro.* .gnu.linkonce.d.rel.ro.*) }
  .dynamic        : { *(.dynamic) }
  .got            : { *(.got) *(.igot) }
  . = DATA_SEGMENT_RELRO_END (SIZEOF (.got.plt) >= 24 ? 24 : 0, .);
  .got.plt        : { *(.got.plt)  *(.igot.plt) }
  .data           :
  {
    *(.data .data.* .gnu.linkonce.d.*)
    SORT(CONSTRUCTORS)
  }
  .data1          : { *(.data1) }
  _edata = .; PROVIDE (edata = .);
  . = .;
  __bss_start = .;
  .bss            :
  {
   *(.dynbss)
   *(.bss .bss.* .gnu.linkonce.b.*)
   *(COMMON)
   /* Align here to ensure that the .bss section occupies space up to
      _end.  Align after .bss to ensure correct alignment even if the
      .bss section disappears because there are no input sections.
      FIXME: Why do we need it? When there is no .bss section, we don't
      pad the .data section.  */
   . = ALIGN(. != 0 ? 64 / 8 : 1);
  }
  .lbss   :
  {
    *(.dynlbss)
    *(.lbss .lbss.* .gnu.linkonce.lb.*)
    *(LARGE_COMMON)
  }
  . = ALIGN(64 / 8);
  . = SEGMENT_START("ldata-segment", .);
  .lrodata   ALIGN(CONSTANT (MAXPAGESIZE)) + (. & (CONSTANT (MAXPAGESIZE) - 1)) :
  {
    *(.lrodata .lrodata.* .gnu.linkonce.lr.*)
  }
  .ldata   ALIGN(CONSTANT (MAXPAGESIZE)) + (. & (CONSTANT (MAXPAGESIZE) - 1)) :
  {
    *(.ldata .ldata.* .gnu.linkonce.l.*)
    . = ALIGN(. != 0 ? 64 / 8 : 1);
  }
  . = ALIGN(64 / 8);
  _end = .; PROVIDE (end = .);
  . = DATA_SEGMENT_END (.);
  /* Stabs debugging sections.  */
  .stab          0 : { *(.stab) }
  .stabstr       0 : { *(.stabstr) }
  .stab.excl     0 : { *(.stab.excl) }
  .stab.exclstr  0 : { *(.stab.exclstr) }
  .stab.index    0 : { *(.stab.index) }
  .stab.indexstr 0 : { *(.stab.indexstr) }
  .comment       0 : { *(.comment) }
  /* DWARF debug sections.
     Symbols in the DWARF debugging sections are relative to the beginning
     of the section so we begin them at 0.  */
  /* DWARF 1 */
  .debug          0 : { *(.debug) }
  .line           0 : { *(.line) }
  /* GNU DWARF 1 extensions */
  .debug_srcinfo  0 : { *(.debug_srcinfo) }
  .debug_sfnames  0 : { *(.debug_sfnames) }
  /* DWARF 1.1 and DWARF 2 */
  .debug_aranges  0 : { *(.debug_aranges) }
  .debug_pubnames 0 : { *(.debug_pubnames) }
  /* DWARF 2 */
  .debug_info     0 : { *(.debug_info .gnu.linkonce.wi.*) }
  .debug_abbrev   0 : { *(.debug_abbrev) }
  .debug_line     0 : { *(.debug_line .debug_line.* .debug_line_end ) }
  .debug_frame    0 : { *(.debug_frame) }
  .debug_str      0 : { *(.debug_str) }
  .debug_loc      0 : { *(.debug_loc) }
  .debug_macinfo  0 : { *(.debug_macinfo) }
  /* SGI/MIPS DWARF 2 extensions */
  .debug_weaknames 0 : { *(.debug_weaknames) }
  .debug_funcnames 0 : { *(.debug_funcnames) }
  .debug_typenames 0 : { *(.debug_typenames) }
  .debug_varnames  0 : { *(.debug_varnames) }
  /* DWARF 3 */
  .debug_pubtypes 0 : { *(.debug_pubtypes) }
  .debug_ranges   0 : { *(.debug_ranges) }
  /* DWARF Extension.  */
  .debug_macro    0 : { *(.debug_macro) }
  .gnu.attributes 0 : { KEEP (*(.gnu.attributes)) }
  /DISCARD/ : { *(.note.GNU-stack) *(.gnu_debuglink) *(.gnu.lto_*) }
}


==================================================
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/crt1.o succeeded
/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/crt1.o
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/crti.o succeeded
/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/crti.o
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/crtbegin.o succeeded
/usr/lib/gcc/x86_64-linux-gnu/4.9/crtbegin.o
attempt to open CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/ColladaSerializer.cpp.o succeeded
CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/ColladaSerializer.cpp.o
attempt to open CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/XmlSerializer.cpp.o succeeded
CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/XmlSerializer.cpp.o
attempt to open CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/util.cpp.o succeeded
CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/util.cpp.o
attempt to open CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/SvgSerializer.cpp.o succeeded
CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/SvgSerializer.cpp.o
attempt to open CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/IfcConvert.cpp.o succeeded
CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/IfcConvert.cpp.o
attempt to open CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/OpenCascadeBasedSerializer.cpp.o succeeded
CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/OpenCascadeBasedSerializer.cpp.o
attempt to open CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/WavefrontObjSerializer.cpp.o succeeded
CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/WavefrontObjSerializer.cpp.o
attempt to open libIfcParse.a succeeded
(libIfcParse.a)IfcUtil.cpp.o
(libIfcParse.a)IfcSpfHeader.cpp.o
(libIfcParse.a)IfcLogger.cpp.o
(libIfcParse.a)IfcSIPrefix.cpp.o
(libIfcParse.a)Ifc2x3.cpp.o
(libIfcParse.a)IfcGlobalId.cpp.o
(libIfcParse.a)IfcParse.cpp.o
(libIfcParse.a)IfcWrite.cpp.o
(libIfcParse.a)Ifc2x3-latebound.cpp.o
(libIfcParse.a)IfcCharacterDecoder.cpp.o
(libIfcParse.a)IfcLateBoundEntity.cpp.o
attempt to open libIfcGeom.a succeeded
(libIfcGeom.a)IfcGeomHelpers.cpp.o
(libIfcGeom.a)IfcGeomRenderStyles.cpp.o
(libIfcGeom.a)IfcGeomFunctions.cpp.o
(libIfcGeom.a)IfcGeomRepresentation.cpp.o
(libIfcGeom.a)IfcRegister.cpp.o
(libIfcGeom.a)IfcGeomMaterial.cpp.o
(libIfcGeom.a)IfcGeomCurves.cpp.o
(libIfcGeom.a)IfcGeomShapes.cpp.o
(libIfcGeom.a)IfcGeomWires.cpp.o
(libIfcGeom.a)IfcGeomFaces.cpp.o
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKernel.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKernel.a failed
attempt to open /usr/local/lib/libTKernel.so succeeded
-lTKernel (/usr/local/lib/libTKernel.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKMath.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKMath.a failed
attempt to open /usr/local/lib/libTKMath.so succeeded
-lTKMath (/usr/local/lib/libTKMath.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKBRep.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKBRep.a failed
attempt to open /usr/local/lib/libTKBRep.so succeeded
-lTKBRep (/usr/local/lib/libTKBRep.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKGeomBase.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKGeomBase.a failed
attempt to open /usr/local/lib/libTKGeomBase.so succeeded
-lTKGeomBase (/usr/local/lib/libTKGeomBase.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKGeomAlgo.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKGeomAlgo.a failed
attempt to open /usr/local/lib/libTKGeomAlgo.so succeeded
-lTKGeomAlgo (/usr/local/lib/libTKGeomAlgo.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKG3d.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKG3d.a failed
attempt to open /usr/local/lib/libTKG3d.so succeeded
-lTKG3d (/usr/local/lib/libTKG3d.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKG2d.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKG2d.a failed
attempt to open /usr/local/lib/libTKG2d.so succeeded
-lTKG2d (/usr/local/lib/libTKG2d.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKShHealing.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKShHealing.a failed
attempt to open /usr/local/lib/libTKShHealing.so succeeded
-lTKShHealing (/usr/local/lib/libTKShHealing.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKTopAlgo.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKTopAlgo.a failed
attempt to open /usr/local/lib/libTKTopAlgo.so succeeded
-lTKTopAlgo (/usr/local/lib/libTKTopAlgo.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKMesh.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKMesh.a failed
attempt to open /usr/local/lib/libTKMesh.so succeeded
-lTKMesh (/usr/local/lib/libTKMesh.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKPrim.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKPrim.a failed
attempt to open /usr/local/lib/libTKPrim.so succeeded
-lTKPrim (/usr/local/lib/libTKPrim.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKBool.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKBool.a failed
attempt to open /usr/local/lib/libTKBool.so succeeded
-lTKBool (/usr/local/lib/libTKBool.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKBO.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKBO.a failed
attempt to open /usr/local/lib/libTKBO.so succeeded
-lTKBO (/usr/local/lib/libTKBO.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKFillet.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKFillet.a failed
attempt to open /usr/local/lib/libTKFillet.so succeeded
-lTKFillet (/usr/local/lib/libTKFillet.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKSTEP.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKSTEP.a failed
attempt to open /usr/local/lib/libTKSTEP.so succeeded
-lTKSTEP (/usr/local/lib/libTKSTEP.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKSTEPBase.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKSTEPBase.a failed
attempt to open /usr/local/lib/libTKSTEPBase.so succeeded
-lTKSTEPBase (/usr/local/lib/libTKSTEPBase.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKSTEPAttr.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKSTEPAttr.a failed
attempt to open /usr/local/lib/libTKSTEPAttr.so succeeded
-lTKSTEPAttr (/usr/local/lib/libTKSTEPAttr.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKXSBase.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKXSBase.a failed
attempt to open /usr/local/lib/libTKXSBase.so succeeded
-lTKXSBase (/usr/local/lib/libTKXSBase.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKSTEP209.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKSTEP209.a failed
attempt to open /usr/local/lib/libTKSTEP209.so succeeded
-lTKSTEP209 (/usr/local/lib/libTKSTEP209.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKIGES.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKIGES.a failed
attempt to open /usr/local/lib/libTKIGES.so succeeded
-lTKIGES (/usr/local/lib/libTKIGES.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKOffset.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libTKOffset.a failed
attempt to open /usr/local/lib/libTKOffset.so succeeded
-lTKOffset (/usr/local/lib/libTKOffset.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libboost_system.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libboost_system.a failed
attempt to open /usr/local/lib/libboost_system.so failed
attempt to open /usr/local/lib/libboost_system.a failed
attempt to open /usr/local/lib64/libboost_system.so failed
attempt to open /usr/local/lib64/libboost_system.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libboost_system.so failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libboost_system.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libboost_system.so succeeded
-lboost_system (/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libboost_system.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libboost_program_options.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libboost_program_options.a failed
attempt to open /usr/local/lib/libboost_program_options.so failed
attempt to open /usr/local/lib/libboost_program_options.a failed
attempt to open /usr/local/lib64/libboost_program_options.so failed
attempt to open /usr/local/lib64/libboost_program_options.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libboost_program_options.so failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libboost_program_options.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libboost_program_options.so succeeded
-lboost_program_options (/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libboost_program_options.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libboost_regex.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libboost_regex.a failed
attempt to open /usr/local/lib/libboost_regex.so failed
attempt to open /usr/local/lib/libboost_regex.a failed
attempt to open /usr/local/lib64/libboost_regex.so failed
attempt to open /usr/local/lib64/libboost_regex.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libboost_regex.so failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libboost_regex.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libboost_regex.so succeeded
-lboost_regex (/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libboost_regex.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libboost_thread.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libboost_thread.a failed
attempt to open /usr/local/lib/libboost_thread.so failed
attempt to open /usr/local/lib/libboost_thread.a failed
attempt to open /usr/local/lib64/libboost_thread.so failed
attempt to open /usr/local/lib64/libboost_thread.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libboost_thread.so failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libboost_thread.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libboost_thread.so succeeded
-lboost_thread (/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libboost_thread.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libboost_date_time.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libboost_date_time.a failed
attempt to open /usr/local/lib/libboost_date_time.so failed
attempt to open /usr/local/lib/libboost_date_time.a failed
attempt to open /usr/local/lib64/libboost_date_time.so failed
attempt to open /usr/local/lib64/libboost_date_time.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libboost_date_time.so failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libboost_date_time.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libboost_date_time.so succeeded
-lboost_date_time (/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libboost_date_time.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libpthread.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libpthread.a failed
attempt to open /usr/local/lib/libpthread.so failed
attempt to open /usr/local/lib/libpthread.a failed
attempt to open /usr/local/lib64/libpthread.so failed
attempt to open /usr/local/lib64/libpthread.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libpthread.so failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libpthread.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libpthread.so succeeded
opened script file /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libpthread.so
opened script file /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libpthread.so
attempt to open /lib/x86_64-linux-gnu/libpthread.so.0 succeeded
/lib/x86_64-linux-gnu/libpthread.so.0
attempt to open /usr/lib/x86_64-linux-gnu/libpthread_nonshared.a succeeded
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libicuuc.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libicuuc.a failed
attempt to open /usr/local/lib/libicuuc.so failed
attempt to open /usr/local/lib/libicuuc.a failed
attempt to open /usr/local/lib64/libicuuc.so failed
attempt to open /usr/local/lib64/libicuuc.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libicuuc.so failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libicuuc.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libicuuc.so succeeded
-licuuc (/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libicuuc.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libicudata.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libicudata.a failed
attempt to open /usr/local/lib/libicudata.so failed
attempt to open /usr/local/lib/libicudata.a failed
attempt to open /usr/local/lib64/libicudata.so failed
attempt to open /usr/local/lib64/libicudata.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libicudata.so failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libicudata.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libicudata.so succeeded
-licudata (/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libicudata.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libdl.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libdl.a failed
attempt to open /usr/local/lib/libdl.so failed
attempt to open /usr/local/lib/libdl.a failed
attempt to open /usr/local/lib64/libdl.so failed
attempt to open /usr/local/lib64/libdl.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libdl.so failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libdl.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libdl.so succeeded
-ldl (/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libdl.so)
attempt to open libIfcParse.a succeeded
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libboost_system.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libboost_system.a failed
attempt to open /usr/local/lib/libboost_system.so failed
attempt to open /usr/local/lib/libboost_system.a failed
attempt to open /usr/local/lib64/libboost_system.so failed
attempt to open /usr/local/lib64/libboost_system.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libboost_system.so failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libboost_system.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libboost_system.so succeeded
-lboost_system (/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libboost_system.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libboost_program_options.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libboost_program_options.a failed
attempt to open /usr/local/lib/libboost_program_options.so failed
attempt to open /usr/local/lib/libboost_program_options.a failed
attempt to open /usr/local/lib64/libboost_program_options.so failed
attempt to open /usr/local/lib64/libboost_program_options.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libboost_program_options.so failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libboost_program_options.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libboost_program_options.so succeeded
-lboost_program_options (/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libboost_program_options.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libboost_regex.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libboost_regex.a failed
attempt to open /usr/local/lib/libboost_regex.so failed
attempt to open /usr/local/lib/libboost_regex.a failed
attempt to open /usr/local/lib64/libboost_regex.so failed
attempt to open /usr/local/lib64/libboost_regex.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libboost_regex.so failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libboost_regex.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libboost_regex.so succeeded
-lboost_regex (/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libboost_regex.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libboost_thread.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libboost_thread.a failed
attempt to open /usr/local/lib/libboost_thread.so failed
attempt to open /usr/local/lib/libboost_thread.a failed
attempt to open /usr/local/lib64/libboost_thread.so failed
attempt to open /usr/local/lib64/libboost_thread.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libboost_thread.so failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libboost_thread.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libboost_thread.so succeeded
-lboost_thread (/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libboost_thread.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libboost_date_time.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libboost_date_time.a failed
attempt to open /usr/local/lib/libboost_date_time.so failed
attempt to open /usr/local/lib/libboost_date_time.a failed
attempt to open /usr/local/lib64/libboost_date_time.so failed
attempt to open /usr/local/lib64/libboost_date_time.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libboost_date_time.so failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libboost_date_time.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libboost_date_time.so succeeded
-lboost_date_time (/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libboost_date_time.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libpthread.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libpthread.a failed
attempt to open /usr/local/lib/libpthread.so failed
attempt to open /usr/local/lib/libpthread.a failed
attempt to open /usr/local/lib64/libpthread.so failed
attempt to open /usr/local/lib64/libpthread.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libpthread.so failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libpthread.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libpthread.so succeeded
opened script file /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libpthread.so
opened script file /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libpthread.so
attempt to open /lib/x86_64-linux-gnu/libpthread.so.0 succeeded
/lib/x86_64-linux-gnu/libpthread.so.0
attempt to open /usr/lib/x86_64-linux-gnu/libpthread_nonshared.a succeeded
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libicuuc.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libicuuc.a failed
attempt to open /usr/local/lib/libicuuc.so failed
attempt to open /usr/local/lib/libicuuc.a failed
attempt to open /usr/local/lib64/libicuuc.so failed
attempt to open /usr/local/lib64/libicuuc.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libicuuc.so failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libicuuc.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libicuuc.so succeeded
-licuuc (/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libicuuc.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libicudata.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libicudata.a failed
attempt to open /usr/local/lib/libicudata.so failed
attempt to open /usr/local/lib/libicudata.a failed
attempt to open /usr/local/lib64/libicudata.so failed
attempt to open /usr/local/lib64/libicudata.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libicudata.so failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libicudata.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libicudata.so succeeded
-licudata (/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libicudata.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libdl.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libdl.a failed
attempt to open /usr/local/lib/libdl.so failed
attempt to open /usr/local/lib/libdl.a failed
attempt to open /usr/local/lib64/libdl.so failed
attempt to open /usr/local/lib64/libdl.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libdl.so failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libdl.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libdl.so succeeded
-ldl (/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libdl.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libstdc++.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libstdc++.a failed
attempt to open /usr/local/lib/libstdc++.so failed
attempt to open /usr/local/lib/libstdc++.a failed
attempt to open /usr/local/lib64/libstdc++.so failed
attempt to open /usr/local/lib64/libstdc++.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libstdc++.so succeeded
-lstdc++ (/usr/lib/gcc/x86_64-linux-gnu/4.9/libstdc++.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libm.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libm.a failed
attempt to open /usr/local/lib/libm.so failed
attempt to open /usr/local/lib/libm.a failed
attempt to open /usr/local/lib64/libm.so failed
attempt to open /usr/local/lib64/libm.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libm.so failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libm.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libm.so succeeded
-lm (/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libm.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libgcc_s.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libgcc_s.a failed
attempt to open /usr/local/lib/libgcc_s.so failed
attempt to open /usr/local/lib/libgcc_s.a failed
attempt to open /usr/local/lib64/libgcc_s.so failed
attempt to open /usr/local/lib64/libgcc_s.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libgcc_s.so succeeded
-lgcc_s (/usr/lib/gcc/x86_64-linux-gnu/4.9/libgcc_s.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libgcc.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libgcc.a failed
attempt to open /usr/local/lib/libgcc.so failed
attempt to open /usr/local/lib/libgcc.a failed
attempt to open /usr/local/lib64/libgcc.so failed
attempt to open /usr/local/lib64/libgcc.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libgcc.so failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libgcc.a succeeded
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libc.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libc.a failed
attempt to open /usr/local/lib/libc.so failed
attempt to open /usr/local/lib/libc.a failed
attempt to open /usr/local/lib64/libc.so failed
attempt to open /usr/local/lib64/libc.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libc.so failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libc.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libc.so succeeded
opened script file /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libc.so
opened script file /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libc.so
attempt to open /lib/x86_64-linux-gnu/libc.so.6 succeeded
/lib/x86_64-linux-gnu/libc.so.6
attempt to open /usr/lib/x86_64-linux-gnu/libc_nonshared.a succeeded
(/usr/lib/x86_64-linux-gnu/libc_nonshared.a)elf-init.oS
attempt to open /lib/x86_64-linux-gnu/ld-linux-x86-64.so.2 succeeded
/lib/x86_64-linux-gnu/ld-linux-x86-64.so.2
/lib/x86_64-linux-gnu/ld-linux-x86-64.so.2
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libgcc_s.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libgcc_s.a failed
attempt to open /usr/local/lib/libgcc_s.so failed
attempt to open /usr/local/lib/libgcc_s.a failed
attempt to open /usr/local/lib64/libgcc_s.so failed
attempt to open /usr/local/lib64/libgcc_s.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libgcc_s.so succeeded
-lgcc_s (/usr/lib/gcc/x86_64-linux-gnu/4.9/libgcc_s.so)
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libgcc.so failed
attempt to open /home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/build/libgcc.a failed
attempt to open /usr/local/lib/libgcc.so failed
attempt to open /usr/local/lib/libgcc.a failed
attempt to open /usr/local/lib64/libgcc.so failed
attempt to open /usr/local/lib64/libgcc.a failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libgcc.so failed
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/libgcc.a succeeded
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/crtend.o succeeded
/usr/lib/gcc/x86_64-linux-gnu/4.9/crtend.o
attempt to open /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/crtn.o succeeded
/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/crtn.o
librt.so.1 needed by /usr/local/lib/libTKernel.so
found librt.so.1 at //lib/x86_64-linux-gnu/librt.so.1
libicui18n.so.52 needed by /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../x86_64-linux-gnu/libboost_regex.so
found libicui18n.so.52 at //usr/lib/x86_64-linux-gnu/libicui18n.so.52
ld-linux-x86-64.so.2 needed by /lib/x86_64-linux-gnu/libpthread.so.0
found ld-linux-x86-64.so.2 at /lib/x86_64-linux-gnu/ld-linux-x86-64.so.2
CMakeFiles/IfcConvert.dir/home/hugo/Documents/dev/ifcopenshell/ifcopenshell-bhb/ifcos/src/ifcconvert/SvgSerializer.cpp.o: In function `SvgSerializer::write(std::pair<std::string, std::vector<util::string_buffer, std::allocator<util::string_buffer> > >&, TopoDS_Wire const&)':
SvgSerializer.cpp:(.text+0x2349): undefined reference to `Geom_Ellipse_Type_()'
SvgSerializer.cpp:(.text+0x2552): undefined reference to `Handle_Standard_Transient::EndScope()'
SvgSerializer.cpp:(.text+0x255c): undefined reference to `Handle_Standard_Transient::EndScope()'
SvgSerializer.cpp:(.text+0x2761): undefined reference to `Handle_Standard_Transient::BeginScope()'
SvgSerializer.cpp:(.text+0x2766): undefined reference to `Geom_Circle_Type_()'
SvgSerializer.cpp:(.text+0x2791): undefined reference to `Handle_Geom_Conic::DownCast(Handle_Standard_Transient const&)'
SvgSerializer.cpp:(.text+0x280f): undefined reference to `Geom_Circle_Type_()'
SvgSerializer.cpp:(.text+0x2835): undefined reference to `Handle_Geom_Ellipse::DownCast(Handle_Standard_Transient const&)'
SvgSerializer.cpp:(.text+0x2869): undefined reference to `Handle_Standard_Transient::EndScope()'
SvgSerializer.cpp:(.text+0x3541): undefined reference to `Handle_Standard_Transient::EndScope()'
SvgSerializer.cpp:(.text+0x3cb1): undefined reference to `Handle_Geom_Circle::DownCast(Handle_Standard_Transient const&)'
SvgSerializer.cpp:(.text+0x3cd8): undefined reference to `Handle_Standard_Transient::EndScope()'
SvgSerializer.cpp:(.text+0x42f2): undefined reference to `TopLoc_SListOfItemLocation::Clear()'
SvgSerializer.cpp:(.text+0x42fb): undefined reference to `Handle_Standard_Transient::EndScope()'
SvgSerializer.cpp:(.text+0x4304): undefined reference to `Handle_Standard_Transient::EndScope()'
...
...
...
IfcGeomFaces.cpp:(.text._ZN24BRepOffsetAPI_MakeOffsetD0Ev[_ZN24BRepOffsetAPI_MakeOffsetD5Ev]+0xaf): undefined reference to `Standard::Free(void*&)'
IfcGeomFaces.cpp:(.text._ZN24BRepOffsetAPI_MakeOffsetD0Ev[_ZN24BRepOffsetAPI_MakeOffsetD5Ev]+0xd0): undefined reference to `BRepFill_ListOfOffsetWire::Clear()'
IfcGeomFaces.cpp:(.text._ZN24BRepOffsetAPI_MakeOffsetD0Ev[_ZN24BRepOffsetAPI_MakeOffsetD5Ev]+0xd9): undefined reference to `TopTools_ListOfShape::Clear()'
IfcGeomFaces.cpp:(.text._ZN24BRepOffsetAPI_MakeOffsetD0Ev[_ZN24BRepOffsetAPI_MakeOffsetD5Ev]+0xfd): undefined reference to `Handle_Standard_Transient::EndScope()'
IfcGeomFaces.cpp:(.text._ZN24BRepOffsetAPI_MakeOffsetD0Ev[_ZN24BRepOffsetAPI_MakeOffsetD5Ev]+0x106): undefined reference to `Handle_Standard_Transient::EndScope()'
IfcGeomFaces.cpp:(.text._ZN24BRepOffsetAPI_MakeOffsetD0Ev[_ZN24BRepOffsetAPI_MakeOffsetD5Ev]+0x11c): undefined reference to `BRepBuilderAPI_Command::Delete()'
IfcGeomFaces.cpp:(.text._ZN24BRepOffsetAPI_MakeOffsetD0Ev[_ZN24BRepOffsetAPI_MakeOffsetD5Ev]+0x137): undefined reference to `TopTools_ListOfShape::Clear()'
IfcGeomFaces.cpp:(.text._ZN24BRepOffsetAPI_MakeOffsetD0Ev[_ZN24BRepOffsetAPI_MakeOffsetD5Ev]+0x15f): undefined reference to `Handle_Standard_Transient::EndScope()'
IfcGeomFaces.cpp:(.text._ZN24BRepOffsetAPI_MakeOffsetD0Ev[_ZN24BRepOffsetAPI_MakeOffsetD5Ev]+0x168): undefined reference to `Handle_Standard_Transient::EndScope()'
collect2: error: ld returned 1 exit status
CMakeFiles/IfcConvert.dir/build.make:250: recipe for target 'IfcConvert' failed
make[2]: *** [IfcConvert] Error 1
CMakeFiles/Makefile2:61: recipe for target 'CMakeFiles/IfcConvert.dir/all' failed
make[1]: *** [CMakeFiles/IfcConvert.dir/all] Error 2
Makefile:117: recipe for target 'all' failed
make: *** [all] Error 2
~~~

--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -142,21 +142,17 @@ ELSE()
 	MESSAGE(STATUS "Looking for opencascade include files in: ${OCC_INCLUDE_DIR}")
 ENDIF()
 
-FIND_FILE(gp_Pnt_hxx "gp_Pnt.hxx" ${OCC_INCLUDE_DIR} /usr/inc /usr/local/inc /usr/local/include/oce)
+FIND_FILE(gp_Pnt_hxx "gp_Pnt.hxx" ${OCC_INCLUDE_DIR})
 IF(gp_Pnt_hxx)
 	MESSAGE(STATUS "Header files found")
 ELSE()
 	MESSAGE(FATAL_ERROR "Unable to find header files, aborting")
 ENDIF()
 
-SET(OPENCASCADE_LIBRARIES
+SET(OPENCASCADE_LIBRARY_NAMES
 	TKernel TKMath TKBRep TKGeomBase TKGeomAlgo TKG3d TKG2d TKShHealing TKTopAlgo TKMesh TKPrim TKBool TKBO
 	TKFillet TKSTEP TKSTEPBase TKSTEPAttr TKXSBase TKSTEP209 TKIGES TKOffset
 )
-
-IF(MSVC)
-	ADD_DEBUG_VARIANTS(OPENCASCADE_LIBRARIES "${OPENCASCADE_LIBRARIES}" "d")
-ENDIF()
 
 IF("${OCC_LIBRARY_DIR}" STREQUAL "")
 	SET(OCC_LIBRARY_DIR "/usr/lib/" CACHE FILEPATH "Open CASCADE library files")
@@ -167,41 +163,53 @@ ELSE()
 	MESSAGE(STATUS "Looking for opencascade library files in: ${OCC_LIBRARY_DIR}")
 ENDIF()
 
-FIND_LIBRARY(libTKernel NAMES TKernel TKerneld PATHS ${OCC_LIBRARY_DIR} /usr/lib /usr/lib64 /usr/local/lib /usr/local/lib64)
+FIND_LIBRARY(libTKernel NAMES TKernel TKerneld PATHS ${OCC_LIBRARY_DIR} NO_DEFAULT_PATH)
 IF(libTKernel)
 	MESSAGE(STATUS "Library files found")
 ELSE()
 	MESSAGE(FATAL_ERROR "Unable to find library files, aborting")
 ENDIF()
 
+# Use the found libTKernel as a template for all other OCC libraries
+foreach(lib ${OPENCASCADE_LIBRARY_NAMES})
+    string(REPLACE TKernel "${lib}" lib_path "${libTKernel}")
+    list(APPEND OPENCASCADE_LIBRARIES "${lib_path}")
+endforeach()
+
+IF(MSVC)
+	ADD_DEBUG_VARIANTS(OPENCASCADE_LIBRARIES "${OPENCASCADE_LIBRARIES}" "d")
+ENDIF()
+
 IF(UNICODE_SUPPORT)
 	# Find ICU 
 	IF("${ICU_INCLUDE_DIR}" STREQUAL "")
 		MESSAGE(STATUS "No ICU include directory specified")
-	ElSE()
-		SET(ICU_INCLUDE_DIR "${ICU_INCLUDE_DIR}" CACHE FILEPATH "ICU header files")
 	ENDIF()
 
 	IF("${ICU_LIBRARY_DIR}" STREQUAL "")
 		MESSAGE(STATUS "No ICU library directory specified")
-	ElSE()
-		SET(ICU_LIBRARY_DIR "${ICU_LIBRARY_DIR}" CACHE FILEPATH "ICU library files")
+		FIND_LIBRARY(icu NAMES icuuc icuucd PATHS ${ICU_LIBRARY_DIR})
+	ELSE()
+		FIND_LIBRARY(icu NAMES icuuc icuucd PATHS ${ICU_LIBRARY_DIR} NO_DEFAULT_PATH)
 	ENDIF()
 
-	FIND_LIBRARY(icu NAMES icuuc icuucd PATHS /usr/lib64 /usr/lib /usr/local/lib64 /usr/local/lib ${ICU_LIBRARY_DIR})
-
 	IF(icu)
+		GET_FILENAME_COMPONENT(ICU_LIBRARY_DIR ${icu} PATH)
+
 		ADD_DEFINITIONS(-DHAVE_ICU)
 		MESSAGE(STATUS "ICU libraries found")
 		# NOTE icudata appears to be icudt on Windows/MSVC and icudata on others
 		#      dl is included to resolve dlopen and friends symbols
 		IF(WIN32)
-			SET(ICU_LIBRARIES icuuc icudt)
+			FIND_LIBRARY(icudt NAMES icudt PATHS ${ICU_LIBRARY_DIR} NO_DEFAULT_PATH)
+			SET(ICU_LIBRARIES ${icu} ${icudt})
 			ADD_DEBUG_VARIANTS(ICU_LIBRARIES "${ICU_LIBRARIES}" "d")
-            # TODO MinGW build would appear to be using dynamic ICU regardless of this definition.
-            ADD_DEFINITIONS(-DU_STATIC_IMPLEMENTATION) # required for static ICU
+			# TODO MinGW build would appear to be using dynamic ICU regardless of this definition.
+			ADD_DEFINITIONS(-DU_STATIC_IMPLEMENTATION) # required for static ICU
 		ELSE()
-			SET(ICU_LIBRARIES icuuc icudata dl)
+			FIND_LIBRARY(icudt NAMES icudata PATHS ${ICU_LIBRARY_DIR} NO_DEFAULT_PATH)
+			FIND_LIBRARY(dl NAMES dl)
+			SET(ICU_LIBRARIES ${icu} ${icudt} ${dl})
 		ENDIF()
 	ELSE()
 		MESSAGE(FATAL_ERROR "UNICODE_SUPPORT enabled, but unable to find ICU. Disable UNICODE_SUPPORT or fix ICU paths to proceed.")
@@ -219,9 +227,13 @@ IF(COLLADA_SUPPORT)
 
 	IF("${OPENCOLLADA_LIBRARY_DIR}" STREQUAL "")
 		MESSAGE(STATUS "No OpenCOLLADA library directory specified")
-		FIND_LIBRARY(OPENCOLLADA_FRAMEWORK_LIB NAMES OpenCOLLADAFramework PATHS /usr/lib64/opencollada /usr/lib/opencollada /usr/lib64 /usr/lib /usr/local/lib64 /usr/local/lib)
+		FIND_LIBRARY(OPENCOLLADA_FRAMEWORK_LIB NAMES OpenCOLLADAFramework
+			PATHS /usr/lib64/opencollada /usr/lib/opencollada /usr/lib64 /usr/lib /usr/local/lib64 /usr/local/lib)
 		GET_FILENAME_COMPONENT(OPENCOLLADA_LIBRARY_DIR ${OPENCOLLADA_FRAMEWORK_LIB} PATH)
 	ENDIF()
+
+	FIND_LIBRARY(OpenCOLLADAFramework NAMES OpenCOLLADAFramework PATHS ${OPENCOLLADA_LIBRARY_DIR} NO_DEFAULT_PATH)
+
 	SET(OPENCOLLADA_LIBRARY_DIR "${OPENCOLLADA_LIBRARY_DIR}" CACHE FILEPATH "OpenCOLLADA library files")
 
 	SET(OPENCOLLADA_INCLUDE_DIRS "${OPENCOLLADA_INCLUDE_DIR}/COLLADABaseUtils" "${OPENCOLLADA_INCLUDE_DIR}/COLLADAStreamWriter")
@@ -230,13 +242,29 @@ IF(COLLADA_SUPPORT)
 	IF(COLLADASWStreamWriter_h)
 		MESSAGE(STATUS "OpenCOLLADA header files found")
 		ADD_DEFINITIONS(-DWITH_OPENCOLLADA)
-		SET(OPENCOLLADA_LIBRARIES
+		SET(OPENCOLLADA_LIBRARY_NAMES
 			GeneratedSaxParser MathMLSolver OpenCOLLADABaseUtils OpenCOLLADAFramework OpenCOLLADASaxFrameworkLoader
-			OpenCOLLADAStreamWriter UTF buffer ftoa pcre
+			OpenCOLLADAStreamWriter UTF buffer ftoa
 		)
-		IF(NOT "${PCRE_LIBRARY_DIR}" STREQUAL "")
+
+		# Use the found libTKernel as a template for all other OCC libraries
+		foreach(lib ${OPENCOLLADA_LIBRARY_NAMES})
+			string(REPLACE OpenCOLLADAFramework "${lib}" lib_path "${OpenCOLLADAFramework}")
+			list(APPEND OPENCOLLADA_LIBRARIES "${lib_path}")
+		endforeach()
+
+		IF("${PCRE_LIBRARY_DIR}" STREQUAL "")
+			find_library(pcre_library NAMES pcre PATHS ${OPENCOLLADA_LIBRARY_DIR})
+			GET_FILENAME_COMPONENT(PCRE_LIBRARY_DIR ${pcre_library} PATH)
+		else()
+			find_library(pcre_library NAMES pcre PATHS ${PCRE_LIBRARY_DIR} NO_DEFAULT_PATH)
+		endif()
+		
+		if (${pcre_library})
 			SET(OPENCOLLADA_LIBRARY_DIR ${OPENCOLLADA_LIBRARY_DIR} ${PCRE_LIBRARY_DIR})
-		ENDIF()
+			list(APPEND OPENCOLLADA_LIBRARIES "${pcre_library}")
+		endif()
+		
 		IF(MSVC)
 			ADD_DEBUG_VARIANTS(OPENCOLLADA_LIBRARIES "${OPENCOLLADA_LIBRARIES}" "d")
 		ENDIF()
@@ -317,9 +345,6 @@ ENDIF()
 INCLUDE_DIRECTORIES(${INCLUDE_DIRECTORIES} ${OCC_INCLUDE_DIR} ${OPENCOLLADA_INCLUDE_DIRS}
 	${ICU_INCLUDE_DIR} ${Boost_INCLUDE_DIRS}
 )
-if(NOT WIN32)
-	INCLUDE_DIRECTORIES(${INCLUDE_DIRECTORIES} /usr/inc /usr/local/inc /usr/local/include/oce)
-endif()
 
 function(files_for_ifc_version IFC_VERSION RESULT_NAME)
     set(IFC_PARSE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../src/ifcparse)
@@ -401,13 +426,6 @@ endif()
 # Boost >= 1.58 requires BOOST_OPTIONAL_USE_OLD_DEFINITION_OF_NONE to build on some Linux distros.
 if(NOT Boost_VERSION LESS 105800)
     add_definitions(-DBOOST_OPTIONAL_USE_OLD_DEFINITION_OF_NONE)
-endif()
-
-LINK_DIRECTORIES(${LINK_DIRECTORIES} ${IfcOpenShell_BINARY_DIR} ${OCC_LIBRARY_DIR} ${OPENCOLLADA_LIBRARY_DIR}
-	${ICU_LIBRARY_DIR} ${Boost_LIBRARY_DIRS}
-)
-if(NOT WIN32)
-	LINK_DIRECTORIES(${LINK_DIRECTORIES} /usr/lib /usr/lib64 /usr/local/lib /usr/local/lib64)
 endif()
 
 if(BUILD_SHARED_LIBS)

--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -335,7 +335,6 @@ IF(MSVC)
 ElSE()
     add_definitions(-Wno-non-virtual-dtor -Wall)
     # TODO Preferably use -Wextra too, but currently too much warning spam coming from the dependencies' headers.
-    add_definitions(-std=c++03)
     # -fPIC is not relevant on Windows and creates pointless warnings
     if (UNIX)
         add_definitions(-fPIC)
@@ -426,6 +425,54 @@ endif()
 # Boost >= 1.58 requires BOOST_OPTIONAL_USE_OLD_DEFINITION_OF_NONE to build on some Linux distros.
 if(NOT Boost_VERSION LESS 105800)
     add_definitions(-DBOOST_OPTIONAL_USE_OLD_DEFINITION_OF_NONE)
+endif()
+
+# Detect OCC version on gcc/clang/mingw as
+# -std=c++11 is needed for OCCT >= 7.0.0
+if(NOT MSVC)
+    FIND_FILE(Standard_Version "Standard_Version.hxx" ${OCC_INCLUDE_DIR})
+
+    set(CMAKE_CONFIGURABLE_FILE_CONTENT "
+        #include <Standard_Version.hxx>
+        #include <iostream>
+        int main(int argc, char** argv) {
+            std::cout << OCC_VERSION_COMPLETE;
+        }")
+    configure_file(
+        "${CMAKE_ROOT}/Modules/CMakeConfigurableFile.in"
+        "${CMAKE_BINARY_DIR}/version.cxx" @ONLY)
+
+    try_compile(VERSION_CHECK
+        ${CMAKE_BINARY_DIR}
+        "${CMAKE_BINARY_DIR}/version.cxx"
+        CMAKE_FLAGS "-DINCLUDE_DIRECTORIES=${OCC_INCLUDE_DIR}"
+        COPY_FILE "${CMAKE_BINARY_DIR}/version"
+        OUTPUT_VARIABLE OUT
+        COPY_FILE_ERROR ERR
+    )
+
+    if(${VERSION_CHECK})
+        EXECUTE_PROCESS(COMMAND ${CMAKE_BINARY_DIR}/version OUTPUT_VARIABLE OCC_VERSION)
+    else()
+        message(FATAL_ERROR "Failed to compile OCC version test:
+        ${OUT}
+        ------
+        ${ERR}")
+    endif()
+
+    MESSAGE(STATUS "OCC version is ${OCC_VERSION}. Detected from: ${Standard_Version}")
+
+    if(NOT ("${OCC_VERSION}" LESS "7.0.0"))
+        include(CheckCXXCompilerFlag)
+        CHECK_CXX_COMPILER_FLAG("-std=c++11" COMPILER_SUPPORTS_CXX11)
+        if(COMPILER_SUPPORTS_CXX11)
+            add_definitions(-std=c++11)
+        else()
+            message(FATAL_ERROR "OCCT7 requires a compiler with C++11 support")
+        endif()
+    else()
+        add_definitions(-std=c++03)
+    endif()
 endif()
 
 if(BUILD_SHARED_LIBS)

